### PR TITLE
Adding timestamps to log output

### PR DESCRIPTION
**Pull request by gmkurtzer**
Adding timestamps per issue request #121.

### CODE REVIEW

File: internal/pkg/warewulfd/daemon.go [hunk 1]
```diff
@@ -48,16 +49,14 @@ func DaemonStart() error {
...
-
 	}
 
 	return nil
 }
 
 func DaemonStatus() error {
 	if !util.IsFile(WAREWULFD_PIDFILE) {
-		wwlog.Printf(wwlog.INFO, "Warewulf daemon process not running (%s)\n", WAREWULFD_PIDFILE)
+		fmt.Printf("Warewulf daemon process not running (%s)\n", WAREWULFD_PIDFILE)
```
Comment (WestleyK) [inline internal/pkg/warewulfd/daemon.go:59]: This should be returning the error. I'v actually already fixed this in #125.

File: internal/pkg/warewulfd/daemon.go [hunk 2]
```diff
@@ -86,7 +85,7 @@ func DaemonStatus() error {
 
 func DaemonReload() error {
 	if !util.IsFile(WAREWULFD_PIDFILE) {
-		wwlog.Printf(wwlog.INFO, "Warewulf daemon process not running (%s)\n", WAREWULFD_PIDFILE)
+		fmt.Printf("Warewulf daemon process not running (%s)\n", WAREWULFD_PIDFILE)
 		return nil
```
Comment (WestleyK) [inline internal/pkg/warewulfd/daemon.go:89]: Again, should return the error.

```suggestion
		return errors.New(fmt.Sprintf("warewulf daemon process is not runnin (%s)", WAREWULFD_PIDFILE))
```

File: internal/pkg/warewulfd/daemon.go [hunk 3]
```diff
@@ -114,7 +113,7 @@ func DaemonReload() error {
 
 func DaemonStop() error {
 	if !util.IsFile(WAREWULFD_PIDFILE) {
-		wwlog.Printf(wwlog.INFO, "Warewulf daemon process not running (%s)\n", WAREWULFD_PIDFILE)
+		fmt.Printf("Warewulf daemon process not running (%s)\n", WAREWULFD_PIDFILE)
 		return nil
```
Comment (WestleyK) [inline internal/pkg/warewulfd/daemon.go:117]: Here too:

```suggestion
		return errors.New(fmt.Sprintf("warewulf daemon process is not runnin (%s)", WAREWULFD_PIDFILE))
```

File: internal/pkg/warewulfd/warewulfd.go [hunk 4]
```diff
@@ -21,30 +20,28 @@ func RunServer() error {
...
-				wwlog.Printf(wwlog.WARN, "Could not load database: %s\n", err)
+				fmt.Printf("ERROR: Could not load database: %s\n", err)
 			}
 		}
 	}()
 
 	err := LoadNodeDB()
 	if err != nil {
-		wwlog.Printf(wwlog.WARN, "Could not load database: %s\n", err)
+		fmt.Printf("ERROR: Could not load database: %s\n", err)
```
Comment (WestleyK) [inline internal/pkg/warewulfd/warewulfd.go:30]: Why not return the error here?

File: internal/pkg/warewulfd/warewulfd.go [hunk 5]
```diff
@@ -21,30 +20,28 @@ func RunServer() error {
...
 	http.HandleFunc("/overlay-runtime", RuntimeOverlaySend)
 
-	wwlog.Printf(wwlog.VERBOSE, "Starting HTTPD REST service\n")
+	daemonLogf("Starting HTTPD REST service\n")
 
 	err = http.ListenAndServe(":9873", nil)
 	if err != nil {
-		wwlog.Printf(wwlog.ERROR, "Could not start listening service: %s\n", err)
+		fmt.Printf("ERROR: Could not start listening service: %s\n", err)
 		os.Exit(1)
```
Comment (WestleyK) [inline internal/pkg/warewulfd/warewulfd.go:45]: Avoid using `os.Exit()`. This should be replaced with: 

```suggestion
		return errors.Wrap(err, "could not start listening service")
```

**Review by WestleyK — CHANGES_REQUESTED** [review]
Looks ok. Some suggestions about error handling.

File: internal/pkg/warewulfd/daemon.go (hunk 1 above)
Comment (gmkurtzer) [reply internal/pkg/warewulfd/daemon.go:59]: It's not an error though, just a status message.

Thoughts on how to best handle this?

Comment (gmkurtzer) [reply internal/pkg/warewulfd/daemon.go]: Fixed in commit update

**Comment by gmkurtzer** [comment]
@WestleyK I think I got all of the suggestions, plus a lot more in terms of how to properly be handling the error conditions and returns. Please check when you get a chance, thanks!

**Comment by gmkurtzer** [comment]
I am merging this to resolve conflicts with Westley's other PRs.

File: internal/app/wwctl/server/status/main.go [hunk 6]
```diff
@@ -2,10 +2,9 @@ package status
 
 import (
 	"github.com/hpcng/warewulf/internal/pkg/warewulfd"
-	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 func CobraRunE(cmd *cobra.Command, args []string) error {
-	return errors.Wrap(warewulfd.DaemonStatus(), "failed to start daemon")
+	return warewulfd.DaemonStatus()
```
Comment (WestleyK) [inline internal/app/wwctl/server/status/main.go:9]: Why did you remove the error wrap? Unless you are using predefined errors (which should be avoided), it is good practice to wrap every error we return.

File: internal/pkg/warewulfd/daemon.go [hunk 7]
```diff
@@ -86,26 +88,22 @@ func DaemonStatus() error {
...
 	}
 
 	pid, _ := strconv.Atoi(string(dat))
 	process, err := os.FindProcess(pid)
-
 	if err != nil {
-		fmt.Printf("Failed to find process: %s\n", err)
-		return err
+		return errors.Wrap(err, "failed to find running PID")
 	} else {
```
Comment (WestleyK) [inline internal/pkg/warewulfd/daemon.go:103]: No need for this `else` statement. _This `else` after return occurs several times here._

**Review by WestleyK — COMMENTED** [review]
I know its already merged, but this can be fixed in another pr.
